eZ/Publish/Core/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php

```diff
@@ -927,7 +928,23 @@ public function loadContentInfo($contentId)
      */
     public function loadContentInfoByRemoteId($remoteId)
     {
-        return $this->internalLoadContentInfo('remote_id', $remoteId);
+        return $this->internalLoadContentInfo('remote_id', 'ezcontentobject', $remoteId);
+    }
+
+    /**
+     * Loads info for a content object identified by its location ID (node ID).
+     *
+     * Returns an array with the relevant data.
+     *
+     * @param mixed $locationId
+     *
+     * @throws \eZ\Publish\Core\Base\Exceptions\NotFoundException
+     *
+     * @return array
+     */
+    public function loadContentInfoByLocationId($locationId)
+    {
+        return $this->internalLoadContentInfo('main_node_id', 'ezcontentobject_tree', $locationId);
```

Comment:
and now going back to usage of arguments inside this method:
`$query->bindValue($id, null, $column === 'id' ? PDO::PARAM_INT : PDO::PARAM_STR)`
we have `main_node_id` which also should be PARAM_INT. Hence my previous comment :)

So let the caller decide on where clause, let's not try to combine it from atomic arguments.

Suggestion: maybe deciding what kind of argument this method needs would be simpler if we used QueryBuilder directly from Doctrine\DBAL, not the Zeta Database components.